src/api/emulations/nfs.js

```diff
@@ -277,20 +279,25 @@ class NFS {
   * @param {Number|CONSTANTS.NFS_FILE_MODE_OVERWRITE|
   *         CONSTANTS.NFS_FILE_MODE_APPEND|
   *         CONSTANTS.NFS_FILE_MODE_READ} [openMode=CONSTANTS.NFS_FILE_MODE_OVERWRITE]
+  * @param {String|Buffer} userMetadata
   * @returns {Promise<File>}
   */
-  open(file, openMode) {
+  open(file, openMode, userMetadata) {
```

Comment:
I'm not so sure about this, it doesn't sound correct to pass new metadata when you are are opening a file, in fact the only case when it's gonna be used is when we are creating a file with this function by passing a `null` file param, otherwise the metadata is ignored without the user noticing it.
Moreover, if I `create` a file, I think the metadata is not used either until I decide to `insert` it into the container (or `update` some other existing one) with some name.

Reply:
Yes, that makes much more sense. Much cleaner as well.